R/estimateParams.R

```diff
@@ -53,3 +53,366 @@ prepareData <- function(data) {
+  } else {
+    negLogLik(transition = weibull_transition(
+      h01 = params[1], h02 = params[2], h12 = params[3],
+      p01 = params[4], p02 = params[5], p12 = params[6]
+    ), data = data)
+  }
+}
+
+#' Format Results of Parameter Estimation for Different Transition Models
+#'
+#' @param transition (`ExponentialTransition` or `WeibullTransition`)\cr
+#'   See [exponential_transition()] or [weibull_transition()] for details.
+#' @param res (`numeric` vector)\cr vector of parameter estimates from the likelihood maximization procedure.
+#'
+#' @return Returns a `TransitionParameters` object with parameter estimates.
+#' @export
+#'
+#' @details
+#' This function dispatches to either `getResults.ExponentialTransition` or `getResults.WeibullTransition`
+#' based on the `transition` object class, using the results from likelihood maximization.
+#'
+#' @examples
+#' results <- c(1.2, 1.5, 1.6)
+#' getResults(exponential_transition(), results)
+getResults <- function(transition, res) {
+  UseMethod("getResults")
+}
+
+#' Format Results of Parameter Estimation for Exponential Transition
+#'
+#' @param transition (`ExponentialTransition`)\cr
+#'   See [exponential_transition()] for details.
+#' @inheritParams getResults
+#'
+#' @return Returns an `ExponentialTransition` object with updated parameter estimates.
+#' @export
+#'
+#' @details
+#' Constructs an `ExponentialTransition` object with parameters estimated from likelihood maximization.
+#'
+#' @examples
+#' results <- c(1.2, 1.5, 1.6)
+#' getResults(exponential_transition(), results)
+getResults.ExponentialTransition <- function(transition, res) {
+  exponential_transition(h01 = res[1], h02 = res[2], h12 = res[3])
+}
+
+#' Format Results of Parameter Estimation for Weibull Transition
+#'
+#' @param transition (`WeibullTransition`)\cr
+#'   See [weibull_transition()] for details.
+#' @inheritParams getResults
+#'
+#' @return Returns a `WeibullTransition` object with updated parameter estimates.
+#' @export
+#'
+#' @details
+#' Constructs a `WeibullTransition` object with parameters estimated from likelihood maximization.
+#'
+#' @examples
+#' results <- c(1.2, 1.5, 1.6, 2, 2.5, 1)
+#' getResults(weibull_transition(), results)
+getResults.WeibullTransition <- function(transition, res) {
+  weibull_transition(
+    h01 = res[1], h02 = res[2], h12 = res[3],
+    p01 = res[4], p02 = res[5], p12 = res[6]
+  )
+}
+
+#' Estimate Parameters of the Multistate Model Using Clinical Trial Data
+#'
+#' @param data (`data.frame`)\cr
+#'   Data frame in the format produced by [getOneClinicalTrial()].
+#' @param transition (`transitionParameters` object)\cr
+#'   Object specifying the assumed distribution of transition hazards.
+#'   Initial parameters for optimization can be specified here.
+#'   See [exponential_transition()] or [weibull_transition()] for details.
+#'
+#' @return Returns a `transitionParameters` object with the estimated parameters.
```

Comment:
```suggestion
#' @return A`TransitionParameters` object with the estimated parameters.
```
(careful with upper vs. lowercase for the class name)